x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -1154,13 +1154,34 @@ public CacheFile getCacheFile(KeyType cacheKey, long length) {
 
     @FunctionalInterface
     public interface RangeAvailableHandler {
-        // caller that wants to read from x should instead do a positional read from x + relativePos
-        // caller should also only read up to length, further bytes will be offered by another call to this method
+        /**
+         * Callback method used to read data from the cache. The target is typically captured by the strategy implementation.
+         *
+         * A caller that wants to read from position p should instead do a positional read from p + relativePos
```

Comment:
I am aware this is an existing comment but I am not able to follow what this trying to say. I think this might also depends on a better explanation on what `relativePos` and how it is provided to the callback?

Reply:
++ this sentence is confusing. I'd just remove it, this is covered (better) by the param descriptions as far as I can tell.